### PR TITLE
Ported PKCS #11 Mutual Auth Demo To Latest MQTT LTS Demo

### PR DESCRIPTION
**Pull request by lundinc2**
Ported PKCS #11 Mutual Auth Demo To Latest MQTT LTS Demo

Description
-----------
* created freertos + mbed tls transport layer implementation using PKCS #11
* Added folder to beta 2 for PKCS #11

Test Steps
-----------
Run the PKCS #11 mutual auth demo

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/mqtt_mutal_auth_with_pkcs11/DemoTasks/MutualAuthMQTTExample.c [hunk 1]
```diff
@@ -0,0 +1,702 @@
...
+ *
+ * The Example shown below uses MQTT APIs to create MQTT messages and send them
+ * over the mutually authenticated network connection established with the
+ * MQTT broker. This example is single threaded and uses statically allocated
+ * memory. It uses QoS1 for sending to and receiving messages from the broker.
+ *
+ * A mutually authenticated TLS connection is used to connect to the
+ * MQTT message broker in this example. Define democonfigMQTT_BROKER_ENDPOINT,
+ * democonfigROOT_CA_PEM, democonfigCLIENT_CERTIFICATE_PEM,
+ * and democonfigCLIENT_PRIVATE_KEY_PEM in demo_config.h to establish a
```
Comment (lundinc2) [inline FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/mqtt_mutal_auth_with_pkcs11/DemoTasks/MutualAuthMQTTExample.c:38]: Update this comment to remove private key and certificate

File: FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/mqtt_mutal_auth_with_pkcs11/main.c [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+const uint8_t ucMACAddress[ 6 ] = { configMAC_ADDR0, configMAC_ADDR1, configMAC_ADDR2, configMAC_ADDR3, configMAC_ADDR4, configMAC_ADDR5 };
+
+/* Use by the pseudo random number generator. */
+static UBaseType_t ulNextRand;
+/*-----------------------------------------------------------*/
+
+int main( void )
+{
+    /***
+     * See https://www.FreeRTOS.org/mqtt_lts/index.html for configuration and usage instructions.
```
Comment (lundinc2) [inline FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/mqtt_mutal_auth_with_pkcs11/main.c:100]: Update to PKCS #11 MQTT demo

File: FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/platform/freertos/transport/src/plaintext_freertos.c [hunk 3]
```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
```
Comment (lundinc2) [inline FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/platform/freertos/transport/src/plaintext_freertos.c:9]: Consider removing this file

File: FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/platform/freertos/transport/src/tls_freertos.c [hunk 4]
```diff
@@ -0,0 +1,972 @@
...
+            mbedtls_ssl_conf_ca_chain( &( pNetworkContext->sslContext.config ),
+                                       &( pNetworkContext->sslContext.rootCa ),
+                                       NULL );
+        }
+    }
+
+    if( returnStatus == TLS_TRANSPORT_SUCCESS )
+    {
+        /* Setup the client private key. */
+        mbedtlsError = initializeClientKeys( &( pNetworkContext->sslContext ) );
```
Comment (lundinc2) [inline FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/platform/freertos/transport/src/tls_freertos.c:307]: update to xResult and make it compare to CKR_OK

File: FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/platform/freertos/transport/src/tls_freertos.c [hunk 5]
```diff
@@ -0,0 +1,972 @@
...
+                                            xClass,
+                                            &xCertObj );
+
+    if( ( CKR_OK == xResult ) && ( xCertObj == CK_INVALID_HANDLE ) )
+    {
+        xResult = CKR_OBJECT_HANDLE_INVALID;
+    }
+
+    /* Query the certificate size. */
+    if( 0 == xResult )
```
Comment (lundinc2) [inline FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/platform/freertos/transport/src/tls_freertos.c:496]: CKR_OK

File: FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/platform/freertos/transport/src/tls_freertos.c [hunk 6]
```diff
@@ -0,0 +1,972 @@
...
+        xResult = pSslContext->pxP11FunctionList->C_GetAttributeValue( pSslContext->xP11Session,
+                                                                       xCertObj,
+                                                                       &xTemplate,
+                                                                       1 );
+    }
+
+    /* Create a buffer for the certificate. */
+    if( 0 == xResult )
+    {
+        xTemplate.pValue = pvPortMalloc( xTemplate.ulValueLen ); /*lint !e9079 Allow casting void* to other types. */
```
Comment (lundinc2) [inline FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/platform/freertos/transport/src/tls_freertos.c:510]: Remove lint comment

File: FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/platform/freertos/transport/src/tls_freertos.c [hunk 7]
```diff
@@ -0,0 +1,972 @@
...
+    /* Decode the certificate. */
+    if( 0 == xResult )
+    {
+        xResult = mbedtls_x509_crt_parse( pxCertificateContext,
+                                          ( const unsigned char * ) xTemplate.pValue,
+                                          xTemplate.ulValueLen );
+    }
+
+    /* Free memory. */
+    if( NULL != xTemplate.pValue )
```
Comment (lundinc2) [inline FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/platform/freertos/transport/src/tls_freertos.c:536]: Don't need this check

File: FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/platform/freertos/transport/src/tls_freertos.c [hunk 8]
```diff
@@ -0,0 +1,972 @@
...
+    {
+        memcpy( &pxCtx->privKeyInfo, mbedtls_pk_info_from_type( xKeyAlgo ), sizeof( mbedtls_pk_info_t ) );
+
+        pxCtx->privKeyInfo.sign_func = privateKeySigningCallback;
+        pxCtx->privKey.pk_info = &pxCtx->privKeyInfo;
+        pxCtx->privKey.pk_ctx = pxCtx;
+    }
+
+    /* Free memory. */
+    if( NULL != pxSlotIds )
```
Comment (lundinc2) [inline FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/pkcs11/platform/freertos/transport/src/tls_freertos.c:666]: redundant check

File: FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/include/tls_freertos_pkcs11.h [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+ * 2. Define LIBRARY_LOG_NAME and  LIBRARY_LOG_LEVEL.
+ * 3. Include the header file "logging_stack.h".
+ */
+
+/* Include header that defines log levels. */
+#include "logging_levels.h"
+
+/* Logging configuration for the Sockets. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME     "TlsTransport"
```
Comment (leegeth) [inline FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/include/tls_freertos_pkcs11.h:48]: please update the name to indicate that this transport uses PKCS as well. There is already another TlsTransport.

Comment (lundinc2) [reply FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/include/tls_freertos_pkcs11.h]: Done

**Pull request merged**